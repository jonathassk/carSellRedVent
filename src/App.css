@import url('https://fonts.googleapis.com/css?family=Saira+Condensed&display=swap');
@media only screen and (max-width: 676px){
  .Container{
  height: 91vh;
  }
}
@media only screen and (min-width: 677px){
  .Container{
  height: 100vh;
  }
}

p{
  font-family: 'Saira Condensed', sans-serif;
}
#containerNav{
  display: flex;
    
  justify-content: center;
  margin: 0px auto;
  padding-top: 40px;
}
button{
  outline: none;
}
body{
  min-width: 350px;
  -webkit-font-smoothing: antialiased;
  margin: 0px;
  padding: 0px; 
}
.Container{
  display: flex;
  justify-content: space-between;
  flex-direction: column;
  position: relative;
  padding: 0px;
  width: 100%;
}
.containerBody{
  height: 450px;
}
a{
  text-decoration: none;
}


