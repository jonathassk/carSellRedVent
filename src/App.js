import React, { Component } from "react";
import "./App.css";
import Item from './Components/item';

export function makeTextWithMistakes(text, mistakes) {
  const paragraphs = text.split("\n");
  const erros = Array(mistakes.length);
  const certos = Array(mistakes.length);
  const empty = '';
  const unidos = [];
  const final = [];

  for(var x = 0; x < mistakes.length; x++){ //separando parte correta de parte errada
      certos[x] = paragraphs[x].slice(0, mistakes[x].start - 1);
      erros[x] = empty.concat(paragraphs[x].slice(mistakes[x].start, mistakes[x].end)) ;
 
  }

  for(x = 0; x < mistakes.length; x++){
    unidos[x] = empty.concat(certos[x],' ...' ,erros[x]); 
    final[x] = unidos[x].split("...")
  };


  const jsonFinal = final.map(item => JSON.parse(JSON.stringify(item)))

  return [jsonFinal];

}
class App extends Component {
  render() {
    const text = "Meu texto est√° erado\nSegundo palagrafo";
    const mistakes = [
      
      {
        start: 15,
        end: 20,
        paragraph: 0
      },
      {
        start: 8,
        end: 17,
        paragraph: 1
      }
    ];

    const errorqt = mistakes.length;
    
    const [errors] = makeTextWithMistakes(text, mistakes); 
    return (
      <div>
       {errors.map((item, key) => <Item key={key} correto={item[0]} errado={item[1]}/>)}
       <div>{errorqt}</div>
      </div>
    );
  }
}

export default App;
