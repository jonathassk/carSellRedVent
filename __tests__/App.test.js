import React from 'react';
import { shallow } from 'enzyme';
import App, {makeTextWithMistakes} from '../src/App';

describe('testando component', () => {
	it('Rendenização deve ser feita', () => {
		const wrapper = shallow(<App />);
		expect(wrapper).toMatchSnapshot();
	});
});

describe('testando separação de paragrafos ', () => {
	test('usando texto com 1 paragrafos', () => {
		expect(makeTextWithMistakes("paragrafo 1")).toEqual(["paragrafo 1"]);
	})
	test('usando texto com 2 paragrafos (texto original do arquivo)', () => {
		expect(makeTextWithMistakes("Meu texto está erado\nSegundo palagrafo")).toEqual(["Meu texto está erado", "Segundo palagrafo"]);
	})
	test('usando texto com 3 paragrafos', () => {
		expect(makeTextWithMistakes("paragrafo 1\nparagrafo 2\nparagrafo 3")).toEqual(["paragrafo 1", "paragrafo 2", "paragrafo 3"]);
	})
})