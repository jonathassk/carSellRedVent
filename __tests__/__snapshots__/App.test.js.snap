// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testando component Rendenização deve ser feita 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Meu texto está erado
</p>,
        <p>
          Segundo palagrafo
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": "Meu texto está erado",
        },
        "ref": null,
        "rendered": "Meu texto está erado",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "Segundo palagrafo",
        },
        "ref": null,
        "rendered": "Segundo palagrafo",
        "type": "p",
      },
    ],
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Meu texto está erado
</p>,
          <p>
            Segundo palagrafo
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": "Meu texto está erado",
          },
          "ref": null,
          "rendered": "Meu texto está erado",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "Segundo palagrafo",
          },
          "ref": null,
          "rendered": "Segundo palagrafo",
          "type": "p",
        },
      ],
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
